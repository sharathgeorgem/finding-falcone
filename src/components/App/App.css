header {
  background-color: black;
  position: static;
  top: 0;
  left: 0;
  padding: 0.5%;
  display: flex;
  justify-content: flex-end;
}

.back {
  font-family: 'Jura', -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
}

footer {
  width: 100%;
  background-color: black;
  position: fixed;
  bottom: 0;
  left: 0;
  padding: 0.5%;
  padding-left: 100%;
}

footer:hover {
  color: #00c367;
  cursor: crosshair;
}

.marquee {
  display: inline-block;
  animation: marquee 15s linear infinite;
}

@keyframes marquee {
  0%   { transform: translate(0, 0); }
  100% { transform: translate(-300%, 0); }
}

.App {
  text-align: center;
}

.container {
  display: grid;
  margin: auto 5%;
  grid-template-columns: repeat(5, 1fr);
  min-height: 80vh;
  grid-gap: 10px;
}

.container img {
  grid-column: 3 / 4;
  border-radius: 10%;
  margin-top: 5%;
}

#container-text {
  grid-column: 2 / 5;
  text-align: justify;
  justify-content: center;
}