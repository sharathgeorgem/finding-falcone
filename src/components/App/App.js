import React, { Component } from 'react'
import { BrowserRouter, Route } from 'react-router-dom'
import './App.css'
import Story from '../Story/Story'
import Planets from '../Planets/Planets'
import Vehicles from '../Vehicles/Vehicles'
import Status from '../Status/Status'

const PLANETS_API = 'https://findfalcone.herokuapp.com/planets'

class App extends Component {
  handleBack = () => {
    window.history.back()
  }
  render () {
    return (
      <BrowserRouter>
        <header>
          <span className='back' onClick={this.handleBack}>Back</span>
        </header>
        <div className='App'>
          <Route exact path='/' component={Story} />
          <Route path='/planets' render={(props) =>
            <Planets {...props} PLANETS_API={PLANETS_API} />} />
          <Route path='/vehicles' component={Vehicles} />
          <Route path='/status' component={Status} />
        </div>
        <footer data-testid='marquee-text'>
          <span className='marquee'>
          Long Live The King of Lengaburu!
          </span>
        </footer>
      </BrowserRouter>
    )
  }
}

export default App
