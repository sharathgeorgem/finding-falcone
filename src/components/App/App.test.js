import React from 'react'
import { render, cleanup } from 'react-testing-library'
import 'jest-dom/extend-expect'
import App from './App'

afterEach(cleanup)

it('renders without crashing', () => {
  const { asFragment } = render(<App text='App component rendered' />)
  expect(asFragment()).toMatchSnapshot()
})

it('inserts text into Header and Footer', () => {
  const { getByTestId, getByText } = render(<App />)
  expect(getByTestId('marquee-text')).toHaveTextContent('Long Live The King of Lengaburu!')
  expect(getByText('Back')).toHaveClass('back')
})
