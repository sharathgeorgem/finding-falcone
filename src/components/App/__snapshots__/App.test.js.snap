// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <header>
    <span
      class="back"
    >
      Back
    </span>
  </header>
  <div
    class="App"
  >
    <div
      class="story"
    >
      <h1>
        Finding Falcone
      </h1>
      <div
        class="storyboard"
      >
        <img
          alt="Falcon"
          src="https://res.cloudinary.com/dmmb5w7sm/image/upload/v1552746277/falcon.png"
        />
        <div
          id="storyboard-text"
        >
          In the planet of Lengaburu...in the distant distant galaxy of Tara B. After the recent war with neighbouring planet Falicornia, King Shan has exiled the Queen of Falicornia for 15 years.
          <br />
          <hr />
          Queen Al Falcone is now in hiding. But if King Shan can find her before the years are up, she will be exiled for another 15 years....
        </div>
        <a
          href="/planets"
          id="nav"
        >
          Where she at?
        </a>
      </div>
    </div>
  </div>
  <footer
    data-testid="marquee-text"
  >
    <span
      class="marquee"
    >
      Long Live The King of Lengaburu!
    </span>
  </footer>
</DocumentFragment>
`;
